R/derive_param_tte.R

```diff
@@ -422,7 +431,7 @@ derive_param_tte <- function(dataset = NULL,
   }
 
   adsl <- dataset_adsl %>%
-    select(!!!adsl_vars)
+    select(!!!adsl_vars, !!!keep_source_vars)
```

Comment:
Using `keep_source_vars` here is different from the other functions. Usually it is used to specify the variables which should be kept from the selected records. To be consistent we would need to add `keep_source_vars` to `event_source()` and `censor_source()`. However, I am not sure if this is a good idea (especially if the default is `exprs(everything())` because different source datasets could be used. For example if `event_source()` uses ADRS and `censor_source()` uses ADSL, all ADRS and all ADSL variables would be added to the new records but the ADSL variables would be populated for censored records only.

Therefore I think we should not add `keep_source_vars` to `derive_param_tte()`.

@bms63 , what do you think?

Reply:
@bms63 @bundfussr I think it's fine if we don't implement it here too, `derive_param_tte()` seems hyperspecific for users, in which case we can just close the PR as well

Reply:
I am fine to not implement this change.  @zdz2101 when you return - can you make a note of this in the function re-work documents.